### PR TITLE
keyspace: Refactor keyspace manager using RunInTxn

### PR DESCRIPTION
**Pull request by AmoebaProtozoa**
### What problem does this PR solve?

Issue Number: Ref #5293

Previous PR #5347 added ability to batch multiple kv operations into a single RunInTxn style execution block, this PR uses that to refactor Keyspace manager. 

In addition, a new keyspace state `tombstone` is added to denote that user data related to keyspace has been completely removed from storage.
 
It also updates kvproto to https://github.com/pingcap/kvproto/pull/1033

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -654,6 +651,10 @@ func (s *Server) bootstrapCluster(req *pdpb.BootstrapRequest) (*pdpb.BootstrapRe
 		log.Warn("flush the bootstrap region failed", errs.ZapError(err))
 	}
 
+	if err := s.GetKeyspaceManager().Bootstrap(); err != nil {
+		log.Warn("bootstrap keyspace manager failed")
```
Comment (disksing) [inline server/server.go:655]: better log the error too.

File: client/keyspace_client.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ type KeyspaceClient interface {
 	LoadKeyspace(ctx context.Context, name string) (*keyspacepb.KeyspaceMeta, error)
 	// WatchKeyspaces watches keyspace meta changes.
 	WatchKeyspaces(ctx context.Context) (chan []*keyspacepb.KeyspaceMeta, error)
+	UpdateKeyspaceState(ctx context.Context, id uint32, state keyspacepb.KeyspaceState) (*keyspacepb.KeyspaceMeta, error)
```
Comment (ystaticy) [inline client/keyspace_client.go:36]: Need to add a comment for UpdateKeyspaceState here?

**Review by disksing — APPROVED** [review]

File: client/keyspace_client.go [hunk 3]
```diff
@@ -111,3 +113,44 @@ func (c *client) WatchKeyspaces(ctx context.Context) (chan []*keyspacepb.Keyspac
...
 	return keyspaceWatcherChan, err
 }
+
+// UpdateKeyspaceState attempts to update the keyspace specified by ID to the target state,
+// it will also record StateChangedAt for the given keyspace if a state change took place.
+// Currently, legal operations includes:
+//
+//	ENABLED -> {ENABLED, DISABLED}
+//	DISABLED -> {ENABLED, DISABLED, ARCHIVED}
+//	ARCHIVED -> {ARCHIVED, TOMBSTONE}
```
Comment (nolouch) [inline client/keyspace_client.go:123]: What the different between the `{ARCHIVED, TOMBSTONE}`?

**Review by nolouch — APPROVED** [review]
LGTM

**Review by disksing — APPROVED** [review]

File: client/keyspace_client.go (hunk 3 above)
Comment (AmoebaProtozoa) [reply client/keyspace_client.go:123]: Archive is used to denote that keyspace's corresponding data in store nodes are safe to remove/ currently being removed (or being archived into cold storage somewhere else), a point of no return for the keyspace (as oppose to disabled which means data are still intact). This tells gc worker to keep cleaning up data belong to this keyspace.

Tombstone is used to denote that keyspace's data in storage has been completely removed, gc worker can safely skip gcing this keyspace

**Pull request merged**